pkg/stanza/fileconsumer/rotation_test.go

```diff
@@ -203,160 +200,57 @@ func TestMultiCopyTruncateSlow(t *testing.T) {
 	wg.Wait()
 }
 
-type rotationTest struct {
-	name            string
-	totalLines      int
-	maxLinesPerFile int
-	maxBackupFiles  int
-	writeInterval   time.Duration
-	pollInterval    time.Duration
-	ephemeralLines  bool
-}
-
-/*
-When log files are rotated at extreme speeds, it is possible to miss some log entries.
-This can happen when an individual log entry is written and deleted within the duration
-of a single poll interval. For example, consider the following scenario:
-  - A log file may have up to 9 backups (10 total log files)
-  - Each log file may contain up to 10 entries
-  - Log entries are written at an interval of 10µs
-  - Log files are polled at an interval of 100ms
-
-In this scenario, a log entry that is written may only exist on disk for about 1ms.
-A polling interval of 100ms will most likely never produce a chance to read the log file.
-
-In production settings, this consideration is not very likely to be a problem, but it is
-easy to encounter the issue in tests, and difficult to deterministically simulate edge cases.
-However, the above understanding does allow for some consistent expectations.
- 1. Cases that do not require deletion of old log entries should always pass.
- 2. Cases where the polling interval is sufficiently rapid should always pass.
- 3. When neither 1 nor 2 is true, there may be missing entries, but still no duplicates.
-
-The following method is provided largely as documentation of how this is expected to behave.
-In practice, timing is largely dependent on the responsiveness of system calls.
-*/
-func (rt rotationTest) expectEphemeralLines() bool {
-	// primary + backups
-	maxLinesInAllFiles := rt.maxLinesPerFile + rt.maxLinesPerFile*rt.maxBackupFiles
-
-	// Will the test write enough lines to result in deletion of oldest backups?
-	maxBackupsExceeded := rt.totalLines > maxLinesInAllFiles
-
-	// last line written in primary file will exist for l*b more writes
-	minTimeToLive := time.Duration(int(rt.writeInterval) * rt.maxLinesPerFile * rt.maxBackupFiles)
-
-	// can a line be written and then rotated to deletion before ever observed?
-	return maxBackupsExceeded && rt.pollInterval > minTimeToLive
-}
+func TestMoveCreate(t *testing.T) {
```

Comment:
It looks like this test `TestMoveCreate` is almost identical to the test `TestMultiFileRotateSlow` above in this file, with the exception of `5ms` vs. `10ms` message write interval. Do they both need to exist?

If I understand correctly, `10ms` interval with 30 messages means that writing to the file exceeds the default `200ms`  file poll interval, while `5s` times 30 messages does not.

Reply:
I'll fix the intervals. The important difference is in the way the files are rotated

Reply:
I've aligned the two tests so that the only difference is the way files are moved.

Reply:
I think we are talking about different tests :upside_down_face: 

I'm comparing `TestMultiFileRotateSlow` with `TestMoveCreate` and diff says they're mostly the same. I think you were maybe looking at `TestCopyTruncate` instead of `TestMultiFileRotateSlow`?

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/70892616/1e330fcf-7899-4ed1-b6e8-3525bb37f446)

Reply:
Ah I see. I renamed that test, not because the behavior had changed, but because I added another test which was very similar. (TestCopyTruncate) The behavior change is not meant to be significant. It's just a clarification that it's testing one of the two common rotation strategies.

Reply:
Yes, both are needed. They are testing different types of file rotation.

Reply:
Awesome, thanks!